ts/test/session/sending/MessageQueue_test.ts

```diff
@@ -0,0 +1,202 @@
+    // Stub out methods which touch the database
+    const storageID = 'pendingMessages';
+    data = {
+      id: storageID,
+      value: '[]',
+    };
+
+    // Pending Message Cache Data Stubs
+    TestUtils.stubData('getItemById')
+      .withArgs('pendingMessages')
+      .callsFake(async () => {
+        return data;
+      });
+    TestUtils.stubData('createOrUpdateItem').callsFake((item: StorageItem) => {
+      if (item.id === storageID) {
+        data = item;
+      }
+    });
+
+    // Utils Stubs
+    sandbox.stub(UserUtil, 'getCurrentDevicePubKey').resolves(ourNumber);
+    TestUtils.stubData('getPairedDevicesFor').callsFake(async () => {
+      return generateMemberList(2);
+    });
+    TestUtils.stubWindow('libsignal', {
+      SignalProtocolAddress: sandbox.stub(),
+      SessionCipher: Stubs.SessionCipherStub,
+    } as any);
+
+    // Message Sender Stubs
+    sendStub = sandbox.stub(MessageSender, 'send').resolves();
+    sendToOpenGroupStub = sandbox
+      .stub(MessageSender, 'sendToOpenGroup')
+      .resolves(true);
+
+    // Group Utils Stubs
+    isMediumGroupStub = sandbox
+      .stub(GroupUtils, 'isMediumGroup')
+      .resolves(false);
+    groupMembersStub = sandbox
+      .stub(GroupUtils, 'getGroupMembers' as any)
+      .callsFake(async () => generateMemberList(10));
+
+    // Session Protocol Stubs
+    hasSessionStub = sandbox.stub(SessionProtocol, 'hasSession').resolves(true);
+    sandbox.stub(SessionProtocol, 'sendSessionRequest').resolves();
+    sandbox
+      .stub(SessionProtocol, 'sendSessionRequestIfNeeded')
+      .callsFake(async (pubkey: PubKey) => {
+        pubkey;
+        sessionRequestSent = true;
+      });
+
+    // Pending Mesage Cache Stubs
+    const chatMessages = Array.from({ length: 10 }, generateChatMessage);
+    const rawMessage = toRawMessage(
+      generateFakePubkey(),
+      generateChatMessage()
+    );
+
+    sandbox.stub(PendingMessageCache.prototype, 'add').resolves(rawMessage);
+    sandbox.stub(PendingMessageCache.prototype, 'remove').resolves();
+    sandbox
+      .stub(PendingMessageCache.prototype, 'getDevices')
+      .returns(generateMemberList(10));
+    sandbox
+      .stub(PendingMessageCache.prototype, 'getForDevice')
+      .returns(chatMessages.map(m => toRawMessage(generateFakePubkey(), m)));
+
+    messageQueueStub = new MessageQueue();
+  });
+
+  afterEach(() => {
+    TestUtils.restoreStubs();
+    sandbox.restore();
+  });
+
+  it('can send to a single device', async () => {
+    const device = generateFakePubkey();
+    const message = generateChatMessage();
+
+    const promise = messageQueueStub.send(device, message);
+    await expect(promise).to.be.fulfilled;
+  });
+
+  it('can send to many devices', async () => {
+    const devices = generateMemberList(10);
+    const message = generateChatMessage();
+
+    const promise = messageQueueStub.sendMessageToDevices(devices, message);
+    await expect(promise).to.be.fulfilled;
+  });
+
```

Comment:
We need tests for the 
```
SyncMessageUtils.canSync
```
code path.

Also these tests seem super basic which for some is alright but for others we need to validate other variables such as a function being called X times or being called with X arguments etc.

For example with the `canSync` a good test would be to check if `sendSyncMessage` was called with the correct arguments if it returns `true`. Otherwise we check if it doesn't get called at all.